### PR TITLE
PFM-ISSUE-7089 - load fe20 applications from via jfrog

### PR DESCRIPTION
**Pull request by smasala**
Grabs the jfrog credentials from gradle for npm

### CODE REVIEW

File: src/model/NPMResolver.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ import {cerr, cgreen, cred, debug, sleepBusy} from "../utils";
 import {PackageVersion} from "./PackageVersion";
 import rimraf = require("rimraf");
 import Timeout = NodeJS.Timeout;
+import { execSync } from "child_process";
```
Comment (slaven3kopic) [inline src/model/NPMResolver.ts:17]: For the consistency reasons I would go with `spawn.sync `(import spawn = require("cross-spawn");)

File: src/model/NPMResolver.ts [hunk 2]
```diff
@@ -164,6 +165,7 @@ export class NPMResolver {
     }
 
     public async resolve(): Promise<void> {
+        this.initRegistry();
```
Comment (slaven3kopic) [inline src/model/NPMResolver.ts:168]: Do we want that on the project level? Or maybe the plugin level?

File: src/model/NPMResolver.ts [hunk 3]
```diff
@@ -323,4 +325,48 @@ export class NPMResolver {
     private getPackageLockPath(): string {
         return path.resolve(this.mainRepo, NPMResolver.PACKAGE_LOCK_JSON);
     }
+
+    private initRegistry(): void {
+        console.info("⟲ Initialising cplace jfrog registry for NPM");
+        const registry = '//cplace.jfrog.io/artifactory/api/npm/cplace-npm-local/';
+        const gradleProps: string = (execSync(path.join(this.mainRepo, 'gradlew properties')) || '').toString();
+
+        if (!gradleProps) {
```
Comment (slaven3kopic) [inline src/model/NPMResolver.ts:334]: Do we want to stop the cplace-asc if anything fails here?

File: src/model/NPMResolver.ts [hunk 4]
```diff
@@ -323,4 +325,48 @@ export class NPMResolver {
...
+                    }
+                }
+            } else {
+                console.error('✗ Gradle / jfrog credentials not found or configured correctly');
+            }
+        }
+    }
+
+    private writeNPMRC(registry, npmrcPath, auth, user) {
+        let npmrc = `@cplace-next:registry=https:${registry} \n`;
```
Comment (slaven3kopic) [inline src/model/NPMResolver.ts:364]: How this works in watch mode?

**Review by slaven3kopic — CHANGES_REQUESTED** [review]
Haven't tested. Code looks good. I wrote just some small concerns/suggestions, but it is up to you. Before merging I would do a test, especially on Windows.

File: src/model/NPMResolver.ts (hunk 2 above)
Comment (smasala) [reply src/model/NPMResolver.ts:168]: It's on npm user level that he credentials are added

File: src/model/NPMResolver.ts (hunk 3 above)
Comment (smasala) [reply src/model/NPMResolver.ts:334]: NPM will fail later due to lack of creds. Unless the user has already logged in. Safe to continue.

File: src/model/NPMResolver.ts (hunk 4 above)
Comment (smasala) [reply src/model/NPMResolver.ts:364]: Only writes if it hasn't done so previously

**Review by slaven3kopic — APPROVED** [review]
Nice!

**Comment by rishoocplace** [comment]
Seems to be failing on windows.
`Error: [cf.cplace.platform] (NPM) installing dependencies failed!
`

**Pull request merged**